draft composite template

jakebeal opened the pull request:
I've put together a draft template for specifying composite parts as well as basic parts. Please look and see if you think that you can incorporate it.

JMante1 commented:
I think it would be nicer to do something like what I have attached (it is more tabular which tends to be easier for people to deal with and this makes it more similar to the basic parts sheet). We might want to at source information too (incase people aren't creating composites simply uploading them from pubmed articles or something). Additionally, if this is reasonable then I can move the examples up so they aren't in the sheet itself and prefill the table only with the false and engineered_region.

I was unclear if commas are a combinatorial part or just several parts being uploaded at once but should all be individual sbol parts?
[template_blank_with_composites_v002.xlsx](https://github.com/SynBioDex/Excel-to-SBOL/files/6038056/template_blank_with_composites_v002.xlsx)

cjmyers commented:
Jet showed me this today, and I concur that making each build a single line looks much cleaner to me.

jakebeal commented:
I think that we can make the horizontal format work, but I found the version that you'd uploaded problematic because it's pushing the parts themselves way off to the side --- on my screen, at least, the parts and the displayID weren't able to be on the screen at the same time, which was bring problematic for me.

I've updated with a slight variation on your proposal that you made. In this version, I've dropped the Role, Target Organism, and Circular columns as redundant.
 - Role and circular (and type too!) should be computed from the parts + backbone + strain
 - When there's a strain, that's the target organism.  When there isn't a strain, target organism is taken from the individual parts (if coherent) and otherwise omitted.

This then leaves most of the screen real estate for the parts. Please let me know what you think.

jakebeal commented:
Separately addressing the question of commas: in my conception, the commas specify a CombinatorialDerivation. Whether that library also gets enumerated is a separate question that could go either way, depending on the workflow.  

For example, if I'm using this as an input for an experiment in SD2 protocol automation, I'd want to interpret it as a CombinatorialDerivation. If I'm specifying a set of parts for a build request, however, they're ultimately going to need to be turned into all of the individual members, whether by this tool or by a downstream tool in the build workflow.

I would thus tend to say that the baseline would be to specify a CombinatorialDerivation, and leave the enumeration of elements for later.

cjmyers commented:
I'm not so sure about role and circular.  For Role, I'm guessing you assume it is always Engineered Region.  However, could it not also be other things such as Terminator when you are specify say a double terminator?

As for circular, I'm not sure how you infer that from the parts.  You can have all linear parts that are composed into a circular plasmid.

jakebeal commented:
Role being engineered region only makes sense when your product is DNA. If your product is a modified strain, then that's a cell type and an NCIT strain role.

As for linear vs. circular: you don't get a circular plasmid without the plasmid backbone, which contains the origin.

cjmyers commented:
Yes, I think you are making my point.  It takes some inference to get these things right, so it likely is good to have the ability to specify them in the table.

jakebeal commented:
Right --- and I believe that I've got them there in the "strain" and "backbone" columns.

jakebeal commented:
I've added a couple more things that are useful for me in the iGEM context, which I'd like your opinions on:
 - A "final product" column on both basic and composite parts, indicating whether in a build order this is intended to be one of the things actually getting shipped, or if it's just an intermediate
 - A constraints part column for combinatorial derivations, e.g., saying two parts are the same or different
 - For composites, modulation annotations RP(X) and X(A,-B) to reverse and trim parts going into composites, respectively.
